postgresql/client.go

```diff
@@ -75,6 +75,19 @@ func NewClient(cfg *Config) *Client {
 		log.Fatal(err)
 	}
 
+	maxAttempts := 20
+	for attempts := 1; attempts <= maxAttempts; attempts++ {
+		err = db.Ping()
+		if err == nil {
+			break
+		}
+		log.Infof("Fail to connect (%d/%d attempts): %v", attempts, maxAttempts, err)
+		time.Sleep(time.Second)
```

Comment:
Here you are sleeping although `attempts == maxAttempts`. I think it is better to do this in a `for {}` loop (without condition) and have the exit condition after the Ping in the loop. So, first try connection and then check whether to fail or retry. Otherwise you will sleep just to fail. So together with the comment above, something like:

```
for {
    err = db.Ping()
    
    if err == nil || attempts++ >= maxAttempts {
         break;
    }

    time.Sleep(time.Second)
}
```